plugins/modules/greppy.py

```diff
@@ -0,0 +1,247 @@
+# i.e. the version is of the form "2.5.0" and not "2.4".
+
+description: Grep logs for a list of patterns within a given timeframe.
+
+options:
+    path:
+        description: The file to grep.
+        required: true
+        type: path
+    search:
+        description: The literal string to look for in every line of the file. This does not have to match the entire line.
+        required: true
+        type: list
+        elements: str
+    exclude:
+        description: The literal string to exclude from the file. This does not have to match the entire line.
+        required: false
+        type: list
+        elements: str
+    timeout:
+        description: The number of seconds to wait.
+        required: false
+        default: 60
+        type: int
+    ignore_case:
+        description: Case insensitive regex search.
+        required: false
+        type: bool
+    findexit:
+        description: Exit right after finding the first occurrence
+        required: false
+        default: false
+        type: bool
+# Specify this value according to your collection
+# in format of namespace.collection.doc_fragment_name
+extends_documentation_fragment:
+    - community.general.greppy
+
+author:
+    - David Peng (@dpengftw)
+'''
+
+EXAMPLES = '''
+- name: Grep for strings in given list but also exclude certain strings
+  community.general.greppy:
+    path: example.txt
+    search:
+      - first string
+      - second string
+    exclude:
+      - third
+      - fourth
+
+- name: Grep for strings and wait for 120 seconds
+  community.general.greppy:
+    path: example.txt
+    search:
+      - first string
+    timeout: 120
+
+- name: Grep for first occurrence then exit
+  community.general.greppy:
+    path: example.txt
+    search:
+      - first occurrence
+    find_exit: yes
+
+'''
+
+RETURN = '''
+# These are examples of possible return values, and in general should use other names for return values.
+output:
+    description: List of lines that matched given search
+    type: list
+    returned: always
+    sample: [
+            "Ex eorum enim scriptis et institutis cum omnis doctrina liberalis, omnis historia.\n",
+            "Esse enim quam vellet iniquus iustus poterat inpune.\n"
+        ]
+
+matches:
+    description: Number of lines that matched
+    type: int
+    returned: always
+    sample: 2
+
+exclude_matches:
+    description: Number of lines that matched the exclusion strings
+    type: int
+    sample: 1
+'''
+
+import re
+import os
+import time
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils._text import to_bytes, to_native
+
+def run_checks(module):
+    """Do baseline checks before proceeding."""
+    params = module.params
+    path = params['path']
+
+    b_dest = to_bytes(path, errors='surrogate_or_strict')
+    if os.path.isdir(b_dest):
+        module.fail_json(rc=256, msg='Destination %s is a directory !' % path)

```

Comment:
That space should only be there in French (and maybe other languages), but not in English.
```suggestion
        module.fail_json(rc=256, msg='Destination %s is a directory!' % path)
```
Another one further below.